Support for SDRplay API version 3.15

fventuri opened the pull request:
I saw that the current version of 'acarsdec' only supports the old SDRplay API (version 2.X), so I rewrote 'sdrplay.c' to support the latest SDRplay API (version 3.15).

I also added a few more options in the SDRplay case to allow to change the sample rate, the IF bandwidth, frequency correction in PPM, and to enable or disable bias-T and the notch filters for BC and DAB that are built in the SDRplay RSP hardware.

Thanks,
Franco

fventuri committed:
SDRplay API version 3.15

f00b4r0 commented:
Hi, thanks for the submission this is a welcome addition. However it's really big. Can you split that huge commit into individual atomic changes? e.g. introducing the V3 API, then adding extra features as separate commits so as to make it easier to review?

Also I'm not sure what's the point of allowing to manually set the sample rate, it's not done for any of the other SDR backends and it adds unnecessary complexity to the code, I'm inclined to NAK that change. Stick to setting the multiplier please. Same with the IF bandwidth, it should be internally computed as is done for RTL and Soapy. Finally please don't unnecessarily rename existing variables (especially to only change character case, I'm talking about `GRdB` here), it adds noise to the changeset review. Thanks!

fventuri commented:
Thanks for the quick reply.
These days I have family visiting, and I'll answer you back after the holidays.

Franco

f00b4r0 commented:
@fventuri ping? :)

fventuri commented:
My apologies Thibaut!
I haven't forgotten about you; I have been so busy with my regular job during the last month that I haven't had much time for the SDR hobby (as a matter of fact I had to work up to few minutes ago).

Hopefully I'll get some time soon, and I'll be able to work on this merge request.

Franco

f00b4r0 commented:
@fventuri hi, any chance you have some time to revisit this soon? I just merged support for Airspy R2, and if you're going to take another look I'll wait before tagging a new release so that we have both changes in :) Thanks

fventuri commented:
@f00b4r0 - I think it is a good idea, and I'll try to find some time to work on it.

I reread your initial comment (it has been awhile since I worked on that change), and I was not sure what you meant by 'split that huge commit into individual atomic changes'.
Do you mean a commit for each file I changed, or a commit for each function I added/changed (if I remember it did a full rewrite since the SDRplay API version 3 is very different from the old version 2).
Of course each single commit alone wouldn't give you a code that works (and probably won't event compile); that's why I am asking.

Franco

f00b4r0 commented:
> @f00b4r0 - I think it is a good idea, and I'll try to find some time to work on it.

Thanks!
 
> I reread your initial comment (it has been awhile since I worked on that change), and I was not sure what you meant by 'split that huge commit into individual atomic changes'. Do you mean a commit for each file I changed, or a commit for each function I added/changed (if I remember it did a full rewrite since the SDRplay API version 3 is very different from the old version 2).

More like a commit for adding v3/removing v2 and then a commit per individual feature you added on top (which you seem to describe in the second paragraph of your first comment). As I mentioned earlier, not all features seem desirable, so having them clearly separated makes evaluating and picking them up easier. It also makes reviewing easier by keeping atomic changes (changes that build of course) together :)

fventuri commented:
I thought about it, and right now I do not really have the time and resources to discuss about features that are "not desirable".
I am going to close this merge request.
If you think it would be useful to have support for v3, please feel free to copy the code from my fork as you see fit.

Franco
